addons/mail/static/src/models/message_action_view.js

```diff
@@ -0,0 +1,205 @@
+                return false;
+            }
+            return clear();
+        },
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeClassAttributes() {
+            const classAttributes = [];
+            classAttributes.push(this.stylingAttribute);
+            switch (this.messageAction.messageActionListOwner) {
+                case this.messageAction.messageActionListOwnerAsDelete:
+                    classAttributes.push('fa fa-lg fa-trash o_MessageActionView_actionDelete');
+                    break;
+                case this.messageAction.messageActionListOwnerAsEdit:
+                    classAttributes.push('fa fa-lg fa-pencil o_MessageActionView_actionEdit');
+                    break;
+                case this.messageAction.messageActionListOwnerAsMarkAsRead:
+                    classAttributes.push('fa fa-lg fa-check o_MessageActionView_actionMarkAsRead');
+                    break;
+                case this.messageAction.messageActionListOwnerAsReaction:
+                    classAttributes.push('fa fa-lg fa-smile-o o_MessageActionView_actionReaction');
+                    break;
+                case this.messageAction.messageActionListOwnerAsReplyTo:
+                    classAttributes.push('fa fa-lg fa-reply o_MessageActionView_actionReplyTo');
+                    break;
+                case this.messageAction.messageActionListOwnerAsToggleCompact:
+                    classAttributes.push('fa fa-lg fa-ellipsis-h o_MessageAction_actionToggleCompact');
+                    break;
+                case this.messageAction.messageActionListOwnerAsToggleStar:
+                    classAttributes.push('fa fa-lg o_MessageActionView_actionToggleStar');
+                    if (this.messageAction.messageActionListOwner.message.isStarred) {
+                        classAttributes.push(`fa-star o_MessageActionView_actionToggleStar_active`);
+                    } else {
+                        classAttributes.push('fa-star-o');
+                    }
+                    break;
+                default:
+                    break;
+            }
+            return classAttributes.join(' ');
+        },
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeStylingAttribute() {
+            const isDeviceSmall = this.messaging.device.isSmall;
+            if (this.messageAction.messageActionListOwner.actionViewsCount == 1) {
+                return isDeviceSmall ? 'ps-3 py-3 pe-3' : 'ps-2 py-2 pe-2';
+            }
+            if (this.messageAction.messageActionListOwner.firstActionView === this) {
+                return isDeviceSmall ? 'ps-3 py-3 pe-2' : 'ps-2 py-2 pe-1';
+            }
+            if (this.messageAction.messageActionListOwner.lastActionView === this) {
+                return isDeviceSmall ? 'ps-2 py-3 pe-3' : 'ps-1 py-2 pe-2';
+            }
+            return isDeviceSmall ? 'px-2 py-3' : 'px-1 py-2';
```

Comment:
It's easier to implement so that we see one bit of padding is linked to a certain condition.

- `ps`:
  - for 1st action, always `3` (small) || `2` (non-small)
  - for non-1st action, always `2` (small) || `1` (non-small)
- `pe`
  - for last action, always `3` (small) || `2` (non-small)
  - for non-last action, always `2` (small) ||  `1` (small)
- `py`
  - always `3` (small) || `2` (non-small)